ee/tabby-webserver/src/service/auth.rs

```diff
@@ -302,6 +289,17 @@ impl AuthenticationService for DbConn {
         Ok(invitation.into())
     }
 
+    async fn request_invitation(&self, email: String) -> Result<Invitation> {
+        if !self
+            .read_security_setting()
+            .await?
+            .can_register_without_invitation(&email)
+        {
+            return Err(anyhow!("Your email does not belong to this organization, please request an invitation from an administrator"));
+        }
+        AuthenticationService::create_invitation(self, email).await
```

Comment:
this won't send invitation email due to the email sending logic is at graphql api level. You'll need to refactor that part a bit.

Reply:
Hmm, not sure the best way to move this to the service level since the auth service can't send mail.